web-api/src/models.rs

```diff
@@ -181,14 +179,16 @@ pub(crate) struct DocumentProperties(HashMap<DocumentPropertyId, DocumentPropert
 
 impl DocumentProperties {
     pub(crate) fn new(
-        properties: HashMap<DocumentPropertyId, DocumentProperty>,
+        mut properties: HashMap<DocumentPropertyId, DocumentProperty>,
         size: usize,
     ) -> Result<Self, InvalidDocumentProperties> {
-        if size <= 2_560 {
-            Ok(Self(properties))
-        } else {
-            Err(InvalidDocumentProperties { size })
+        if size > 2_560 {
+            return Err(InvalidDocumentProperties { size });
```

Comment:
This size will include nullable properties we do remove AFIK, but we can still merge it and then fix that later.